src/style/style.js

```diff
@@ -985,17 +985,20 @@ class Style extends Evented {
             this._layerOrderChanged = false;
         }
 
-        if (!this.pauseablePlacement.isDone()) {
+        if (this.pauseablePlacement.isDone()) {
+            // the last placement finished running, but the next one hasn’t
+            // started yet because of the `stillRecent` check immediately
+            // above, so mark it stale to ensure that we request another
+            // render frame
+            this.placement.setStale();
+        } else {
             this.pauseablePlacement.continuePlacement(this._order, this._layers, layerTiles);
 
             if (this.pauseablePlacement.isDone()) {
-                const placement = this.pauseablePlacement.placement;
-                placementChanged = placement.commit(this.placement, browser.now());
-                if (!this.placement || placementChanged || symbolBucketsChanged) {
-                    this.placement = placement;
-                    this.collisionIndex = this.placement.collisionIndex;
-                }
-                this.placement.setRecent(browser.now(), placement.stale);
+                this.pauseablePlacement.placement.commit(this.placement, browser.now());
```

Comment:
_"placement placement placement placement!"_ 😉 

This is already a great improvement, but maybe we can go further here by giving `PausablePlacement` its own commit wrapper that handles some of this state tracking? Ideally these three lines could be one line.